.semaphore/semaphore.yml

```diff
@@ -19,60 +19,164 @@ queue:
+              value: stable
+      epilogue: &build-test-epilogue
         always:
           commands:
             - make remove-test-env
-  - name: "Build & Test (VS Code Insiders)"
+            - make ci-bin-sem-cache-store
+            - make store-test-results-to-semaphore
+
+  - name: "Build & Test (VS Code: linux arm64)"
     dependencies: []
-    skip:
-      when: "branch != 'main' AND change_in(['/release.svg', '/.versions/next.txt'], {pipeline_file: 'ignore', branch_range: '$SEMAPHORE_GIT_COMMIT_RANGE', default_branch: 'main'})"
+    skip: *build-test-skip-stable
+    task:
+      agent:
+        machine:
+          type: s1-prod-ubuntu24-04-arm64-1
+      prologue: *build-test-prologue
+      jobs: *build-test-jobs-stable
+      epilogue: *build-test-epilogue
+
+  # Disabled until we can connect to an active GUI session
+  # - name: "Build & Test (VS Code: darwin x64)"
+  #   dependencies: []
+  #   skip: *build-test-skip-stable
+  #   task:
+  #     agent:
+  #       machine:
+  #         type: s1-prod-macos-13-5-amd64
+  #     prologue: *build-test-prologue
+  #     jobs: *build-test-jobs-stable
+  #     epilogue: *build-test-epilogue
+
+  # - name: "Build & Test (VS Code: darwin arm64)"
+  #   dependencies: []
+  #   skip: *build-test-skip-stable
+  #   task:
+  #     agent:
+  #       machine:
+  #         type: s1-prod-macos-13-5-arm64
+  #     prologue: *build-test-prologue
+  #     jobs: *build-test-jobs-stable
+  #     epilogue: *build-test-epilogue
+
+  - name: "Build & Test (VS Code: win32 x64)"
+    dependencies: []
+    skip: *build-test-skip-stable
     task:
+      agent:
+        machine:
+          type: s1-prod-windows
+      prologue:
+        commands:
+          - checkout
+          - make ci-bin-sem-cache-restore
+          - $Env:PATH += ";C:\Program Files\Git\bin"
+          - $Env:PATH += ";C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64"
+          # Ensure npm and npx are available
+          - npm --version
+          - npx --version
+          - choco install gh -y
+          - $Env:PATH += ";C:\Program Files\GitHub CLI\"
+          - gh --version
+          - |
+            if (Test-Path "$HOME/.gh-token") {
+              $Env:GH_TOKEN = Get-Content "$HOME/.gh-token" -Raw | ForEach-Object { $_.Trim() }
+              if ([string]::IsNullOrEmpty($Env:GH_TOKEN)) {
+                  Write-Error "Token file exists but is empty"
+                  exit 1
+              }
+              gh config set prompt disabled
+              $Env:GH_TOKEN | gh auth login --with-token
+              gh auth status
+            } else {
+                Write-Error "Token file not found at $HOME/.gh-token"
+                exit 1
+            }
+          - |
+            if ($Env:SEMAPHORE_ORGANIZATION_URL -like "*.semaphoreci.com") {
+              Write-Output "Skipping Vault setup for Semaphore CI"
+            } else {
+              $Env:VAULT_ADDR = "https://vault.cireops.gcp.internal.confluent.cloud"
```

Comment:
Do we have to set this manually or is it not the default? I'd rather we not include such URLs in the `main` branch of a public repo, but I guess it's fine if we _really_ need to.

Reply:
🤦  Never mind -- https://github.com/confluentinc/ide-sidecar/blob/main/.semaphore/multi-arch-builds-and-upload.yml#L116

Reply:
I guess we do have to set it this way. Should be fine.. right?!